src/library_async.js

```diff
@@ -723,6 +724,9 @@ mergeInto(LibraryManager.library, {
             asyncFinalizers.forEach(function(func) {
               func(asyncWasmReturnValue);
             });
+            Asyncify.sleepCallbacks.forEach(function(func) {
```

Comment:
This never works in dosbox case,  cause condition on line 709 **never** became true.
```
          if (!Asyncify.currData) {
```
![image](https://user-images.githubusercontent.com/1727152/68831043-f9676d00-06df-11ea-88bd-2b56410046df.png)
if I move `Asyncify.sleepCallbacks.forEach` before this everything is fixed. Actually I don't know what meaning of this currData.

Reply:
Even this test code didn't work:
```
#include <SDL/SDL.h>
#include <stdio.h>
#include <sys/time.h>
#include <algorithm>
using namespace std;

#ifdef EMSCRIPTEN
#include <emscripten.h>
#endif

long int reportedSec = 0;
struct timeval startTime;
int available = 0;

void func_callback(void *unused, Uint8 *stream, int len) {

    for (int i=0;i<std::min(len, available);i++) {
        stream[i] = i;
    }

    for (int i = available; i < len; ++i) {
        stream[i] = 0;
    }
    available = 0;
}

void loop() {
    struct timeval nowTime;
    gettimeofday(&nowTime, NULL);

    if (nowTime.tv_sec != reportedSec) {
        reportedSec = nowTime.tv_sec;
        printf("playing %ld sec...\n", reportedSec - startTime.tv_sec);
    }

    for (int i = 0; i < 15; ++i) {
        available = 1024;
        #ifdef EMSCRIPTEN
            //EM_ASM(({
            //    SDL.audio.queueNewAudioData();
            //}));
            emscripten_sleep(10);
        #endif
    }
}

int main(int argc, char *argv[])
{
    gettimeofday(&startTime, NULL);
    if( SDL_Init(SDL_INIT_TIMER | SDL_INIT_AUDIO ) <0 ) {
        SDL_WM_SetCaption( "Something busy..", NULL );
        return 1;
    }

    SDL_AudioSpec as;

    as.freq = 22000;
    as.format = AUDIO_U8;
    as.samples = 1024;
    as.callback=func_callback;
    as.userdata=NULL;
    as.channels = 1;
    
    if ( SDL_OpenAudio(&as, NULL) < 0 ) {
        SDL_WM_SetCaption( "Unable to open audio..", NULL );
        return 1;
    }

    SDL_Surface *screen = SDL_SetVideoMode(200,200, 16, SDL_SWSURFACE);
    
    SDL_PauseAudio(0);

#ifdef EMSCRIPTEN
    emscripten_set_main_loop(loop, 0, true);
#endif

    return 0;
}
```

Reply:
Thanks @caiiiycuk ! Yeah, that code was not in the right place, I realize now.

I pushed an update, please tell me if that fixes it.

I tested your testcase here, and it prints "playing 0", 1, 2, 3 etc., and buzzes, does that mean it works?

Reply:
@kripken Yeah this fix works for dosobox and testcase. Thanks!

--
Testcase works if it constantly buzzes without hangs.